changematrix/src/main/java/it/geosolutions/jaiext/changematrix/ChangeMatrixOpImage.java

```diff
@@ -176,6 +181,18 @@ public ChangeMatrixOpImage(final RenderedImage reference, final RenderedImage no
         // Area grid is present
         areaGrid = area != null;
         areaMap = area;
+        
+        // is NoData range present?
+        this.hasNoData = (noData!=null)?true:false;
+        this.noData = noData;
+        if(this.hasNoData){
+            // Check if No ChangeMatrix classes set belong to NoData
+            for(Integer clazz : result.getRegisteredClasses()){
+                if(noData.contains(clazz)){
+                    throw new IllegalStateException("One or more provided classes are contained in the NoData range");
+                }
+            }
+        }
     }
```

Comment:
For new lines of code, make sure to use proper formatting templates.
if(condition){ 

should be

if (condition) {

same for for and so on.